File: implementation/scipamato/scipamato-core-web/src/test/java/ch/difty/scipamato/core/web/common/SelfUpdateEventTest.java
package ch.difty.scipamato.core.web.common;

import static org.assertj.core.api.Assertions.assertThat;

import org.apache.wicket.ajax.AjaxRequestTarget;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
public class SelfUpdateEventTest {

    @Mock
    private AjaxRequestTarget targetMock;

    @Test
    public void instantiate() {
        SelfUpdateEvent e = new SelfUpdateEvent(targetMock);
        assertThat(e.getTarget()).isEqualTo(targetMock);
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/common/SelfUpdateEventTest.kt
package ch.difty.scipamato.core.web.common

import ch.difty.scipamato.common.AjaxRequestTargetSpy
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test

internal class SelfUpdateEventTest {

    @Test
    fun instantiate() {
        val targetDummy = AjaxRequestTargetSpy()
        val e = SelfUpdateEvent(targetDummy)
        e.target shouldBeEqualTo targetDummy
    }
}
